Add observability samples (self managed) for Istio addon (#4744)
Azure/AKS: examples/istio-based-service-mesh/observability/README.md

# Telemetry samples

> The addon already have native integration with Azure managed Prometheus and managed Grafana etc. We strongly suggest Azure customers use these managed solutions.

Reviewer: ```suggestion
The addon already has native integration with Azure managed Prometheus and managed Grafana etc. We strongly recommend Azure customers use these managed solutions.
```
